stixcore/products/level0/science.py

```diff
@@ -286,11 +289,11 @@ def from_levelb(cls, levelb):
 
         counts_var = np.sqrt(counts_var.transpose((0, 2, 3, 1)))
         if ssid == 21:
-            out_counts = np.zeros((unique_times.size, 32, 12, 32))
-            out_var = np.zeros((unique_times.size, 32, 12, 32))
+            out_counts = np.zeros((unique_times.size, 32, data['num_pixel_sets'][0], 32))
+            out_var = np.zeros((unique_times.size, 32, data['num_pixel_sets'][0], 32))
         elif ssid == 22:
-            out_counts = np.zeros((unique_times.size, 32, 4, 32))
-            out_var = np.zeros((unique_times.size, 32, 4, 32))
+            out_counts = np.zeros((unique_times.size, 32, data['num_pixel_sets'][0], 32))
+            out_var = np.zeros((unique_times.size, 32, data['num_pixel_sets'][0], 32))
```

Comment:
how will the packet merge (same fits file) behave with different data['num_pixel_sets'] ?

Reply:
Eh good question it will break at the moment I think another reason to store all BSD on a per request basis and not OBT day but open to other suggestions?

Reply:
ah i thought for BSD it will be by request anyway.